### PR TITLE
Add workloadidentityuser Service connection support to PipAuthenticateV1

### PR DESCRIPTION
**Pull request by embetten**
**Task name**: PipAuthenticateV1

**Description**: Add workloadidentityuser Service connection support to PipAuthenticateV1

**Documentation changes required:** (Y/N) Y

**Added unit tests:** (Y/N) N

**Attached related issue:** (Y/N) N

**Checklist**:
- [X] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [X] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/PipAuthenticateV1/_buildConfigs/Node20/package.json [hunk 1]
```diff
@@ -24,11 +24,11 @@
   },
   "homepage": "https://github.com/Microsoft/azure-pipelines-tasks#readme",
   "dependencies": {
-    "@types/node": "^20.3.1",
     "@types/mocha": "^5.2.7",
+    "@types/node": "^20.3.1",
     "@types/uuid": "^8.3.0",
-    "azure-pipelines-tasks-artifacts-common": "^2.212.0",
-    "azure-pipelines-task-lib": "^4.0.0-preview"
+    "azure-pipelines-task-lib": "^4.0.0-preview",
```
Comment (starkmsu) [inline Tasks/PipAuthenticateV1/_buildConfigs/Node20/package.json:30]: Please update azure-pipelines-task-lib as well

File: Tasks/PipAuthenticateV1/package.json [hunk 2]
```diff
@@ -24,11 +24,11 @@
     },
     "homepage": "https://github.com/Microsoft/azure-pipelines-tasks#readme",
     "dependencies": {
-        "@types/node": "^16.11.39",
         "@types/mocha": "^5.2.7",
+        "@types/node": "^16.11.39",
         "@types/uuid": "^8.3.0",
-        "azure-pipelines-tasks-artifacts-common": "^2.212.0",
-        "azure-pipelines-task-lib": "^4.0.0-preview"
+        "azure-pipelines-task-lib": "^4.0.0-preview",
```
Comment (starkmsu) [inline Tasks/PipAuthenticateV1/package.json:30]: Please update azure-pipelines-task-lib as well

File: Tasks/PipAuthenticateV1/task.json [hunk 3]
```diff
@@ -42,6 +42,27 @@
         "MultiSelectFlatList": "True"
       }
     },
+    {
+      "name": "workloadIdentityServiceConnection",
```
Comment (starkmsu) [inline Tasks/PipAuthenticateV1/task.json:46]: Do we want to support ONLY WIF service connections? Should it be possible to use other types of ARM service connections?

File: Tasks/PipAuthenticateV1/pipauthenticatemain.ts [hunk 4]
```diff
@@ -13,9 +14,33 @@ async function main(): Promise<void> {
...
     try {
-        let internalAndExternalEndpoints: string[] = [];
+        const feedUrl = tl.getInput("feedUrl");
+        const entraWifServiceConnectionName = tl.getInput("workloadIdentityServiceConnection");
 
+        // If using feed Url and wif service connection, set the pip index url
+        if (feedUrl && entraWifServiceConnectionName)
+        {
+            tl.debug(tl.loc("Info_AddingFederatedFeedAuth", entraWifServiceConnectionName, feedUrl));
+            const feedTenant = await getFeedTenantId(feedUrl);
```
Comment (starkmsu) [inline Tasks/PipAuthenticateV1/pipauthenticatemain.ts:27]: feedTenant might be null or undefined here - we need to check for that

File: Tasks/PipAuthenticateV1/_buildConfigs/Node20/package.json (hunk 1 above)
Comment (embetten) [reply Tasks/PipAuthenticateV1/_buildConfigs/Node20/package.json]: done

File: Tasks/PipAuthenticateV1/task.json (hunk 3 above)
Comment (embetten) [reply Tasks/PipAuthenticateV1/task.json:46]: at this point yes, this is the design we have been working with. Happy to discuss further offline

File: Tasks/PipAuthenticateV1/pipauthenticatemain.ts (hunk 4 above)
Comment (embetten) [reply Tasks/PipAuthenticateV1/pipauthenticatemain.ts:27]: the feed tenant can be null but is being passed as an optional parameter, if unresolved it falls back to the service connection tenant. This approach mimics our other cred providers and would unblock multi tenanted apps.

**Review by merlynomsft — APPROVED** [review]

**Pull request merged**